Source/Core/Core/AchievementManager.cpp

```diff
@@ -1262,22 +1275,31 @@ u32 AchievementManager::MemoryPeeker(u32 address, u8* buffer, u32 num_bytes, rc_
 {
   if (buffer == nullptr)
     return 0u;
+#ifdef RC_CLIENT_SUPPORTS_RAINTEGRATION
+  auto& instance = AchievementManager::GetInstance();
+  std::lock_guard lg{instance.m_memory_lock};
+  u32 bytes_read = 0;
+  for (u32 ix = address; ix < address + num_bytes && ix < instance.m_cloned_memory.size(); ix++)
+    buffer[bytes_read++] = instance.m_cloned_memory[ix];
+  return bytes_read;
+#else   // RC_CLIENT_SUPPORTS_RAINTEGRATION
```

Comment:
MemoryPeeker is now going to behave differently depending on whether you have RC_CLIENT_SUPPORTS_RAINTEGRATION defined. With it defined, the data returned will be up to one frame old, and you can only read from MEM1. Without it defined, the data returned will be fresh, and you can read from all memory regions.

Unless you can be sure that these differences don't matter (maybe when running in non-development mode, MemoryPeeker is only ever called in connection to DoFrame, and no achievement developer will ever try to read from anything but MEM1?), what this will mean in practice is that non-Windows builds are going to have different behavior from Windows builds and set developers will have no way to test the non-Windows behavior. Please make sure you have thought this through.